lib/event_store_client/configuration.rb

```diff
@@ -1,28 +1,47 @@
 # frozen_string_literal: true
 
-require 'dry-struct'
-require 'singleton'
+require 'dry-configurable'
 
 module EventStoreClient
-  class Configuration
-    include Singleton
+  extend Dry::Configurable
 
-    attr_accessor :per_page, :service_name, :mapper, :error_handler, :pid_path, :adapter
+  # Supported adapters: %i[api in_memory]
+  #
+  setting :adapter, :api
 
-    def configure
-      yield(self) if block_given?
-    end
+  setting :error_handler
+  setting :eventstore_url, 'http://localhost:2113' do |value|
+    value.is_a?(String) ? URI(value) : value
```

Comment:
**Question**: Why do you need it here?

Reply:
I want to ensure it's a valid URL. Working with strings makes no sense, as the app will probably crash for invalid data. URI ensures it's at least parsable URL wich reduces the risk of errors. However, there is a bug here. I should have the condition reversed: `valude.is_a?(URI) ? value : URI(value)` - this way config will raise an error for invalid data.